client/db/bolt/db.go

```diff
@@ -1624,6 +1624,129 @@ func (idx *timeIndexNewest) add(t uint64, k []byte, b *bbolt.Bucket) {
 	})
 }
 
+// DeleteInactiveOrders deletes orders that no longer needed for normal
+// operations. Optionally accepts a time to delete orders with a later time
+// stamp.
+func (db *BoltDB) DeleteInactiveOrders(olderThan *time.Time) (deletedOrders []*dexdb.MetaOrder, finished bool, err error) {
+	start := time.Now()
+	const batchSize = 1000
+	var olderThanB []byte
+	if olderThan != nil {
+		olderThanB = uint64Bytes(encode.UnixMilliU(*olderThan))
+	} else {
+		olderThanB = uint64Bytes(encode.UnixMilliU(start))
+	}
+	deletedOrders = make([]*dexdb.MetaOrder, 0, batchSize)
+	return deletedOrders, finished, db.Update(func(tx *bbolt.Tx) error {
+		// Some archived orders may still be needed for active matches.
+		// Put those order id's in a map to prevent deletion.
+		amb := tx.Bucket(activeMatchesBucket)
+		activeMatches := make(map[order.OrderID]struct{})
+		amb.ForEach(func(k, _ []byte) error {
+			mBkt := amb.Bucket(k)
+			if mBkt == nil {
+				return fmt.Errorf("match %x bucket is not a bucket", k)
```

Comment:
Looks like this error is ignored. (`ForEach` error ignored)

Reply:
Returning now if error.